src/App.tsx

```diff
@@ -21,42 +45,45 @@ export const App: React.FC = () => {
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!user && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {isLoading ? (
+                  <Loader />
+                ) : (
+                  <>
+                    {isPostsExist && (
+                      <div
+                        className="notification is-warning"
+                        data-cy="NoPostsYet"
+                      >
+                        No posts yet
+                      </div>
+                    )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                    {isPostListShown && (
+                      <PostsList />
+                    )}
+                  </>
```

Comment:
You are reading the posts in the App only to show this section

Possibly it makes sense to move the block "No posts yet" to the `PostsList`